rust_src/remacs-sys/lib.rs

```diff
@@ -20,6 +20,62 @@ include!(concat!(env!("OUT_DIR"), "/definitions.rs"));
 
 pub type Lisp_Object = EmacsInt;
 
+pub const PSEUDOVECTOR_SIZE_BITS: libc::c_int = 12;
+pub const PSEUDOVECTOR_SIZE_MASK: libc::c_int = (1 << PSEUDOVECTOR_SIZE_BITS) - 1;
+pub const PSEUDOVECTOR_REST_BITS: libc::c_int = 12;
+pub const PSEUDOVECTOR_REST_MASK: libc::c_int = (((1 << PSEUDOVECTOR_REST_BITS) - 1) <<
+                                                 PSEUDOVECTOR_SIZE_BITS);
+pub const PSEUDOVECTOR_AREA_BITS: libc::c_int = PSEUDOVECTOR_SIZE_BITS + PSEUDOVECTOR_REST_BITS;
+pub const PVEC_TYPE_MASK: libc::c_int = 0x3f << PSEUDOVECTOR_AREA_BITS;
+
+pub type pvec_type = libc::c_int;
+pub const PVEC_NORMAL_VECTOR: pvec_type = 0;
+pub const PVEC_FREE: pvec_type = 1;
+pub const PVEC_PROCESS: pvec_type = 2;
+pub const PVEC_FRAME: pvec_type = 3;
+pub const PVEC_WINDOW: pvec_type = 4;
+pub const PVEC_BOOL_VECTOR: pvec_type = 5;
+pub const PVEC_BUFFER: pvec_type = 6;
+pub const PVEC_HASH_TABLE: pvec_type = 7;
+pub const PVEC_TERMINAL: pvec_type = 8;
+pub const PVEC_WINDOW_CONFIGURATION: pvec_type = 9;
+pub const PVEC_SUBR: pvec_type = 10;
+pub const PVEC_OTHER: pvec_type = 11;
+pub const PVEC_XWIDGET: pvec_type = 12;
+pub const PVEC_XWIDGET_VIEW: pvec_type = 13;
+pub const PVEC_COMPILED: pvec_type = 14;
+pub const PVEC_CHAR_TABLE: pvec_type = 15;
+pub const PVEC_SUB_CHAR_TABLE: pvec_type = 16;
+pub const PVEC_FONT: pvec_type = 17;
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct vectorlike_header {
+    pub size: libc::ptrdiff_t,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct Lisp_Subr {
+    pub header: vectorlike_header,
+    // This field is actually an union
+    pub function: *const libc::c_void,
+    pub min_args: libc::c_short,
+    pub max_args: libc::c_short,
+    pub symbol_name: *const libc::c_char,
+    pub intspec: *const libc::c_char,
+    // In, Emacs C, this is an EMACS_INT. Should this be changed?
```

Comment:
Wow, well spotted! Apparently this used to be a `*char`: https://github.com/emacs-mirror/emacs/commit/c5461d03a411ff5c6f43885a0a9030e8a94bbc2e which is why it probably works.

Worth adding a `TODO` I think.